src/main/kotlin/com/hibob/ex2-loops.kt

```diff
@@ -0,0 +1,40 @@
+package com.hibob
+
+fun isContainsSpecialCharacterExcludingUnderscore(s: String): Boolean {
+    val regex = "[^a-zA-Z0-9_]".toRegex()
+    return regex.containsMatchIn(s)
+}
+
+fun isContainOnlyNumbers(s: String): Boolean {
+    var numberOnly = true
+    for (char in s) {
+        if (char in '0'..'9') continue else numberOnly = false
+    }
+    return !numberOnly
+}
+
+fun isValidIdentifier(s: String): Boolean {
+    // vacant
+    if (s == "" ) {
+        return false
+    }
+
+    // is first char start with number
+    if (s[0] in '0'..'9') {
+        return false
+    }
+
+    val res = isContainOnlyNumbers(s)
```

Comment:
This will always return `false`, because you got to this point after the check in line 23, which means that you already know that `s[0]` is not a number.